carbon-feature-plugin/src/main/java/org/wso2/maven/p2/utils/ProductFileUtils.java

```diff
@@ -0,0 +1,159 @@
+import org.wso2.maven.p2.beans.carbon.product.config.ProductConfig;
+import org.wso2.maven.p2.beans.carbon.product.config.ProductFileConfig;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+
+/**
+ * Utility class for helper functions in generating and reading a product file at build-time.
+ *
+ * This class is to be used by publish-product and
+ * generate-runtime maven goals.
+ *
+ * @since 3.1.0
+ */
+public class ProductFileUtils {
+
+    /**
+     * Write Product bean as an XML to a product file.
+     *
+     * @param productFileConfig productFileConfig bean as read from project pom.
+     * @param mavenProject mavenProject bean as read from project pom.
+     * @throws javax.xml.bind.JAXBException Is thrown if an error occurs in writing configurations into product file.
+     */
+    public static void writeToFile(ProductFileConfig productFileConfig,
+                                   MavenProject mavenProject) throws JAXBException, MojoExecutionException {
+        // validating version and featureConfig values captured from project pom
+        ProductConfig validatedProductConfig = ProductFileUtils.
+                validateProductConfigForVersionAndFeatureConfig(productFileConfig.getProductConfig(), mavenProject);
+
+        // populating a product instance with validated values for writing to a product file
+        Product product = new Product();
+
+        product.setName(validatedProductConfig.getName());
+        product.setUid(validatedProductConfig.getUid());
+        product.setId(validatedProductConfig.getId());
+        product.setApplication(validatedProductConfig.getApplication());
+        product.setVersion(validatedProductConfig.getVersion());
+        product.setUseFeatures(validatedProductConfig.getUseFeatures());
+        product.setIncludeLaunchers(validatedProductConfig.getIncludeLaunchers());
+        product.setVersion(validatedProductConfig.getVersion());
+        product.setConfigIni(validatedProductConfig.getConfigIni());
+        product.setLauncherArgs(validatedProductConfig.getLauncherArgs());
+        product.setLauncher(validatedProductConfig.getLauncher());
+        product.setPlugins(validatedProductConfig.getPlugins());
+
+        Features features = new Features(validatedProductConfig.getFeatureConfig());
+        Configurations configurations = new Configurations(validatedProductConfig.getPluginConfig(),
+                validatedProductConfig.getPropertyConfig());
+
+        product.setFeatures(features);
+        product.setConfigurations(configurations);
+
+        // writing to a product file
+        JAXBContext jaxbContext = JAXBContext.newInstance(Product.class);
+        Marshaller jaxbMarshaller = jaxbContext.createMarshaller();
+
+        jaxbMarshaller.setProperty("com.sun.xml.internal.bind.xmlHeaders",
```

Comment:
I think the correct fix is to use the jaxb-impl dependency [1], and then set the property "com.sun.xml.bind.xmlHeaders" instead of the internal property. 

PR is good to go after this change.

[1] http://search.maven.org/#artifactdetails%7Cjavax.xml%7Cjaxb-impl%7C2.1%7Cjar